src/WallpaperEngine/Data/Parsers/ObjectParser.cpp

```
@@ -101,6 +101,51 @@ SoundUniquePtr ObjectParser::parseSound (const JSON& it, ObjectData base) {
+    const auto& properties = project.properties;
+    const auto textIt = it.require ("text", "Text object must have a text field");
+
+    std::string text;
+    std::string script;
+
+    if (textIt.is_string ()) {
+	text = textIt.get<std::string> ();
+    } else if (textIt.is_object ()) {
+	// dynamic text with a script; Phase 1 does not execute the script, so
+	// just capture the source and leave the rendered text empty
+	if (const auto scriptIt = textIt.optional<std::string> ("script"); scriptIt.has_value ()) {
```

Cppcheck (2.20.0)

[error] Found an exit path from function with non-void return type that has missing return statement

(missingReturn)